### ISSUE TITLE
{Linked:Bug13586388}[Adaptive Cards > Designer] - Screen Reader is not conveying the descriptive information while navigating on the 'Book this trip' button.

### ISSUE THREAD

**Issue by vagpt** [microsoft/AdaptiveCards#9163]
### Target Platforms

Other

### SDK Version

1.6

### Application Name

Adaptive Cards

### Problem Description

**Test Environment:**
URL: https://adaptivecards.io/designer/
OS Version: 24H2 (OS Build 27766.1000)
Browser Version: Version 132.0.2186.2 (Official build) dev (64-bit)
Screen Reader: NVDA

**Pre-Requisite:**
Run the below attached JSON file under 'Copy Card Payload editor.'

**Repro Steps:**
1. Open the above URL on the Edge Dev browser.
2. Press tab key and navigate to the 'Book this trip' button.
3. Now again press tab key navigate to the 'Book this trip' button and observe the issue.

**Actual Result:**
While navigating on the 'Book this trip' button, screen reader is not conveying the descriptive information. Here. it is announcing as **'Book this trip button'**.

**Expected Result:**
Screen Reader should announce the descriptive information while navigating on the 'Book this trip' button for the first button it should announce as **'24-7 Test Route AM, Apr 24' Book this trip button'** and for second it should announce as **'24-7 Test Route AM, Apr 25 Book this trip button**'.

**Note:**
Same issue is repro with Narrator screen reader as well.

**User Impact:**
It will impact the screen reader user as they will not be able to differentiate between both buttons if the screen reader will not announce the descriptive information. Here user will be confused and not able to know on which button focus is present.

**WCAG Reference:**
https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships

Have feedback on bugs logged, please tag bug as “A11yRCA” and add your feedback in the comment section.

### Screenshots

![Image](https://github.com/user-attachments/assets/6ef5bc42-cead-4eb3-af8a-f587955fa1ae)

https://github.com/user-attachments/assets/f9458dc9-612b-48f5-a659-403ee52a068e

### Card JSON

```JSON
{
  "type": "AdaptiveCard",
  "version": "1.5",
  "speak": "First trip. 24-7 Test Route AM, Apr 24. Depart from REDMOND TECHNOLOGY STATION (RTS) at 02:01 PM. Arrive at 10TH AVE E & E ALOHA ST at 02:15 PM.",
  "body": [
    {
      "type": "Container",
      "spacing": "medium",
      "separator": true,
      "items": [
        {
          "type": "TextBlock",
          "size": "medium",
          "weight": "bolder",
          "text": "24-7 Test Route AM, Apr 24",
          "wrap": true,
          "style": "heading",
          "spacing": "small"
        },
        {
          "type": "TextBlock",
          "weight": "lighter",
          "text": "Depart from",
          "wrap": true
        },
        {
          "type": "ColumnSet",
          "spacing": "small",
          "columns": [
            {
              "type": "Column",
              "width": "stretch",
              "items": [
                {
                  "type": "TextBlock",
                  "text": "REDMOND TECHNOLOGY STATION (RTS)",
                  "wrap": true
                }
              ]
            },
            {
              "type": "Column",
              "width": "auto",
              "items": [
                {
                  "type": "TextBlock",
                  "text": "02:01 PM",
                  "horizontalAlignment": "right",
                  "wrap": true
                }
              ]
            }
          ]
        },
        {
          "type": "TextBlock",
          "weight": "lighter",
          "text": "Arrive at",
          "wrap": true
        },
        {
          "type": "ColumnSet",
          "spacing": "small",
          "columns": [
            {
              "type": "Column",
              "width": "stretch",
              "items": [
                {
                  "type": "TextBlock",
                  "text": "10TH AVE E & E ALOHA ST",
                  "wrap": true
                }
              ]
            },
            {
              "type": "Column",
              "width": "auto",
              "items": [
                {
                  "type": "TextBlock",
                  "text": "02:15 PM",
                  "horizontalAlignment": "right",
                  "wrap": true
                }
              ]
            }
          ]
        },
        {
          "type": "ActionSet",
          "actions": [
            {
              "type": "Action.Submit",
              "title": "Book this trip",
              "data": {
                "id": "trip1"
              }
            }
          ]
        }
      ]
    },
    {
      "type": "Container",
      "spacing": "medium",
      "separator": true,
      "items": [
        {
          "type": "TextBlock",
          "size": "medium",
          "weight": "bolder",
          "text": "24-7 Test Route AM, Apr 25",
          "wrap": true,
          "style": "heading",
          "spacing": "small"
        },
        {
          "type": "TextBlock",
          "weight": "lighter",
          "text": "Depart from",
          "wrap": true
        },
        {
          "type": "ColumnSet",
          "spacing": "small",
          "columns": [
            {
              "type": "Column",
              "width": "stretch",
              "items": [
                {
                  "type": "TextBlock",
                  "text": "REDMOND TECHNOLOGY STATION (RTS)",
                  "wrap": true
                }
              ]
            },
            {
              "type": "Column",
              "width": "auto",
              "items": [
                {
                  "type": "TextBlock",
                  "text": "04:01 PM",
                  "horizontalAlignment": "right",
                  "wrap": true
                }
              ]
            }
          ]
        },
        {
          "type": "TextBlock",
          "weight": "lighter",
          "text": "Arrive at",
          "wrap": true
        },
        {
          "type": "ColumnSet",
          "spacing": "small",
          "columns": [
            {
              "type": "Column",
              "width": "stretch",
              "items": [
                {
                  "type": "TextBlock",
                  "text": "10TH AVE E & E ALOHA ST",
                  "wrap": true
                }
              ]
            },
            {
              "type": "Column",
              "width": "auto",
              "items": [
                {
                  "type": "TextBlock",
                  "text": "06:15 PM",
                  "horizontalAlignment": "right",
                  "wrap": true
                }
              ]
            }
          ]
        },
        {
          "type": "ActionSet",
          "actions": [
            {
              "type": "Action.Submit",
              "title": "Book this trip",
              "data": {
                "id": "trip2"
              }
            }
          ]
        }
      ]
    }
  ]
}
```

**Comment by vagpt** [comment]
Originally tracking with https://microsoftit.visualstudio.com/OneITVSO/_workitems/edit/13586388

**Comment by SatwikKrSharma** [comment]
@vagpt it is working as expected in the designer, please check again

**Comment by SatwikKrSharma** [comment]
Hi @vagpt it is working as expected as per the accessibility rules because for the first section only the property is set ""speak": "First trip. 24-7 Test Route AM, Apr 24. Depart from REDMOND TECHNOLOGY STATION (RTS) at 02:01 PM. Arrive at 10TH AVE E & E ALOHA ST at 02:15 PM.","
for the second button no such information is provided so the narrator does not know what to convey.
hence this is not an accessibility bug

**Comment by SatwikKrSharma** [comment]
With the first party windows narrator it is working as expected

**Comment by vagpt** [comment]
As per the discussion with Dev team it is working as expected with Narrator and repro'ing with NVDA only then PG team needs to log 3P bug for this, hence closing this issue.
